### DISCUSSION TITLE
App crashing on M1

### DISCUSSION THREAD

**Discussion by rfperuch** [The-Wineskin-Project/WineskinServer#281] [General]
Hi there!

I'm trying to port a game to an M1 machine but I'm facing the following error output in LastRunWine.log

`assertion failed [(_unoccupied_temporary_gprs & mask) != 0]: invalid temporary request
(TranslatorHelpers.cpp:105 allocate_temporary_gpr_num)
`
The port works very well on Intel machines but it refuses to start on M1. The macOS is the latest (Monterey). 
I've tried to find that assertion on Crossover's Source code but I didn't find it. 

Does anyone have a clue on how to fix it?

**Comment by Gcenx** [comment]
> The port works very well on Intel machines but it refuses to start on M1.

Going off what you’ve said yourself it’s working fine on Intel but not on M1 indicating the problem is Rosetta2.

**Reply by rfperuch** [reply]
Hi @Gcenx thanks for the answer!

Do you know how I can debug a game ported with Wineskin/wine running in the Rosetta2 translation environment?

**Reply by rfperuch** [reply]
This is the whole LastRun.log file

`0024:fixme:seh:get_thread_times not implemented on this platform
0024:fixme:ntdll:NtQuerySystemInformation (0x00000067,0x31ec7c,0x00000008,0x31ec84) stub
assertion failed [(_unoccupied_temporary_gprs & mask) != 0]: invalid temporary request
(TranslatorHelpers.cpp:105 allocate_temporary_gpr_num)
`
